tests/conftest.py

```diff
@@ -328,30 +313,87 @@ def _navigate_function(text=None, use_clipboard=False):
 
 @pytest.fixture(name="find_telemetry")
 def fixture_find_telemetry(selenium):
-    def _(ping, ping_data=None, scalar_type="keyedScalars"):
-        stored_events = []
+    def _(ping, scalar=None, value=None, scalar_type="keyedScalars"):
         control = True
-        timeout = time.time() + 60 * 2
+        timeout = time.time() + 60
 
         while control and time.time() < timeout:
-            telemetry = requests.get(f"{PING_SERVER}/pings").json()
-            for event in telemetry:
-                try:
-                    stored_events.append(event["payload"]["processes"]["parent"][scalar_type])
-                except KeyError:
-                    pass
-                else:
-                    continue
-            for item in stored_events:
-                data = item.get(ping)
-                if data is not None and ping_data == data:
+            match scalar_type:
+                case "keyedScalars":
+                    script = """
+                        return Services.telemetry.getSnapshotForKeyedScalars()
+                    """
+                    with selenium.context(selenium.CONTEXT_CHROME):
+                        telemetry = selenium.execute_script(script)
+                    try:
+                        for item, val in telemetry["parent"].get(ping).items():
+                            if scalar == item and value == val:
+                                logging.info(f"Parent Pings {telemetry['parent']}\n")
+                                return True
+                    except (TypeError, AttributeError):
+                        continue
+                case "scalars":
+                    script = """
+                            return Services.telemetry.getSnapshotForScalars()
+                        """
+                    with selenium.context(selenium.CONTEXT_CHROME):
+                        telemetry = selenium.execute_script(script)
+                    assert telemetry["parent"].get(ping) == value
+                    logging.info(f"Parent Pings {telemetry['parent']}\n")
```

Comment:
Q: does this log need to be any different from the logging on L331 so we know which path is logging?

Reply:
Different how? These will return different logs since the script that feeds the ```telemetry``` variable is different